tests/integration/s2n_tls13_handshake_tests.py

```diff
@@ -47,7 +50,25 @@ def verify_hrr_random_data(server, client):
 
     return result
 
+def key_update_recv(server, client):
+  
+    result = Result()
+    result.status = Status.PASSED
+    openssl_msg = "Message:" + str(uuid.uuid4())
+    if client.args[0] == "openssl":
+        client.stdin.write(("k\n\n").encode("utf-8"))
+        client.stdin.flush()
+        time.sleep(1)
```

Comment:
have you figured out why sleep(1) is needed?

Reply:
Not yet.

Reply:
During my testing I had to sleep() to give the openssl process time to read the data I'd written. Otherwise all the data written after the 'k' would be dropped. "All the data" meaning `openssl_msg` below.

After a keyupdate, s_client sets a buffer length to 0, which looks like it causes the next write to be of 0 length. So the keyupdate command needs to be in a separate read() from the rest of the data.

Reply:
I was able to remove the sleep statement by waiting until s_client outputted a KEYUPDATE statement.